File: Algorithms/search_sort/Java/binarysearch.java
public class binarysearch {

    public static int search(int[] arr, int value){

        int n = arr.length,i = 0, j = n - 1, m = (i + j) / 2;
        for(; i <= j; m = (i + j) / 2){
            if(arr[m] == value)
                return m;
            else if(arr[m] < value)
                i = m + 1;
            else
                j = m - 1;
        }
        return -1;
    }
}

File: Algorithms/search_sort/Python/binarysearch.py

def search(arr, key):

    left, right = 0, len(arr) - 1
    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == key:
            return mid
        elif arr[mid] > key:
            right = mid - 1
        else:
            left = mid + 1
    return -1
